src/components/CommentList.jsx

```diff
@@ -0,0 +1,51 @@
+import { getComments } from "../utils"
+import { CommentListItem } from "./"
+import { useState, useEffect } from "react"
+
+
+export const CommentList = (props) => {
+
+    const {articleId} = props
+
+    const [result, setResult] = useState([])
+    const [loading, setLoading] = useState(true)
+    const [error, setError] = useState(false)
+  
+    const errMsg = "Something went wrong loading comments, try reloading the page" 
+    //in theory this error should be impossible to trigger with user error as an invalid article ID would be caught in its parent
+
+    useEffect(() => {
+        setError(false)
+        setLoading(true)
+        
+        getComments(articleId)
+        .then((comments) => {
+            setLoading(false)
+            setResult(comments)
+        })
+        .catch((e) => {
+            setLoading(false)
+            setError(true)
+        })
+    }, [])
+
+    if (error) {
+        return <p className="warning-text">{errMsg}</p>
+    } else if (loading) {
+        return <p>Please wait while comments load</p>
+    } 
+
```

Comment:
What should appear if an article has no comments?

Reply:
Good point - I should invite the user to be the first to comment of course! now added